lib/src/main/java/com/orange/ods/compose/component/chip/OdsChip.kt

```diff
@@ -0,0 +1,139 @@
+ *
+ * @param text Text to display in the chip.
+ * @param onClick called when the chip is clicked.
+ * @param modifier Modifier to be applied to the chip
+ * @param outlined If set to true, a border will be drawn around the chip.
+ * @param enabled When disabled, chip will not respond to user input. It will also appear visually
+ * disabled and disabled to accessibility services.
+ * @param leadingIcon Optional icon at the start of the chip, preceding the content text.
+ * @param leadingAvatar Optional avatar at the start of the chip, preceding the content text.
+ * @param leadingElementContentDescription Content description associated to the leading element (icon or avatar).
+ * @param onCancel called when the cancel cross of the chip is clicked. Pass `null` here for no cancel cross.
+ */
+@ExperimentalMaterialApi
+@Composable
+fun OdsChip(
+    text: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    outlined: Boolean = false,
+    enabled: Boolean = true,
+    selected: Boolean = false,
+    leadingIcon: Painter? = null,
+    leadingAvatar: Painter? = null,
+    leadingElementContentDescription: String? = null,
+    onCancel: (() -> Unit)? = null
+) {
+    Chip(
+        onClick = onClick,
+        modifier = modifier,
+        border = if (outlined) {
+            BorderStroke(1.dp, odsChipBorderColor(selected = selected, enabled = enabled))
+        } else null,
+        enabled = enabled,
+        colors = odsChipColors(outlined, selected),
+        leadingIcon = when {
+            leadingIcon != null -> {
+                {
+                    Icon(
+                        modifier = Modifier.size(dimensionResource(id = R.dimen.chip_icon_size)),
+                        painter = leadingIcon,
+                        contentDescription = leadingElementContentDescription
+                    )
+                }
+            }
+            leadingAvatar != null -> {
+                {
+                    OdsImageCircleShape(
+                        painter = leadingAvatar,
+                        contentDescription = leadingElementContentDescription,
+                        circleSize = dimensionResource(id = R.dimen.icon_size),
+                        alpha = if (enabled) 1f else LeadingIconOpacity
+                    )
+                }
+            }
+            else -> null
+        }
+    ) {
+        OdsTextBody2(text = text)
+        onCancel?.let {
+            Spacer(modifier = Modifier.size(dimensionResource(id = R.dimen.ods_spacing_xs)))
+            Icon(
+                modifier = Modifier
+                    .size(18.dp)
+                    .clickable { onCancel() },
+                painter = painterResource(id = R.drawable.ic_cancel),
+                contentDescription = "cancel",
+                tint = odsChipColors(outlined, selected).leadingIconContentColor(enabled = enabled).value
```

Comment:
There is a square background when cancel icon is clicked. Should we remove this effect ?

![Screenshot_20220629_162450](https://user-images.githubusercontent.com/13621389/176461107-102aac9c-0703-406f-99fc-7b1887bf0b31.png)

Reply:
Waw how did you see that?! :D